### PR TITLE
CTCP-1885: Search for LRN

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**

### CODE REVIEW

File: app/controllers/departure/drafts/DashboardController.scala [hunk 1]
```diff
@@ -16,37 +16,72 @@
...
+        .fold(
+          formWithErrors => buildView(formWithErrors, pageSize)(BadRequest(_)),
+          lrn => {
+            val fuzzyLrn: Option[String] = Option(lrn).filter(_.trim.nonEmpty)
+            buildView(form, pageSize, fuzzyLrn, isSearch = true)(Ok(_))
+          }
+        )
+  }
+
+  private def buildView(form: Form[String], pageSize: Int, lrn: Option[String] = None, isSearch: Boolean = false)(
```
Comment (jackheslop96) [inline app/controllers/departure/drafts/DashboardController.scala:69]: Is `isSearch` not just always equal to `lrn.isDefined` ?

File: app/controllers/departure/drafts/DashboardController.scala [hunk 2]
```diff
@@ -16,37 +16,72 @@
...
+  )(implicit request: IdentifierRequest[_]): Future[Result] = {
+
+    val getDrafts = lrn match {
+      case Some(value) => draftDepartureService.getLRNs(value, pageSize)
+      case None        => draftDepartureService.getAll()
+    }
+
+    getDrafts.map {
+      case Some(drafts) =>
+        val toViewModel = AllDraftDeparturesViewModel(drafts)
```
Comment (jackheslop96) [inline app/controllers/departure/drafts/DashboardController.scala:80]: Could you pass the `lrn` and `pageSize` into the `AllDraftDeparturesViewModel` `apply` method and do the `tooManyResults` method in the case class? Then you only need to pass the form and view model into the view

File: app/controllers/departure/drafts/DashboardController.scala (hunk 1 above)
Comment (jamesMaxwell191) [inline app/controllers/departure/drafts/DashboardController.scala:69]: Maybe flags are better avoided (as in a function should do one thing). Something to consider

File: app/controllers/departure/drafts/DashboardController.scala (hunk 2 above)
Comment (jamesMaxwell191) [reply app/controllers/departure/drafts/DashboardController.scala:80]: I would say that is the way to go (all logic in the view model )

**Review by jackheslop96 — DISMISSED** [review]

**Review by jamesMaxwell191 — APPROVED** [review]

**Review by kenneth-rayner — DISMISSED** [review]

**Pull request merged**